### PR TITLE
added outfit add and remove

### PR DESCRIPTION
**Pull request by hangyin2020**
added a button to add outfit, will change it to a img card. The remove button now can remove the current item in the outfit

### CODE REVIEW

File: client/src/components/RelatedItems/OutfitItemsList.jsx [hunk 1]
```diff
@@ -1,26 +1,72 @@
...
+    const outfitItemsCopy = [...outfitItems];
+    // check if current item is already in outfit
+    let currentItemInOutfit = false;
+    outfitItemsCopy.forEach((item) => {
+      if (item.id === itemId) {
+        currentItemInOutfit = true;
+      }
+    });
+    if (!currentItemInOutfit) { outfitItemsCopy.push(currentItem); }
+    setOutfitItems(outfitItemsCopy);
```
Comment (slargman) [inline client/src/components/RelatedItems/OutfitItemsList.jsx:35]: shorter way:
```
let currentItemInOutfit = outfitItems.any((item) => item.id === itemId);
if (!currentItemInOutfit) { setOutfitItems([...outfitItems, currentItem]); }
```

**Review by slargman — APPROVED** [review]
Looks good!

File: client/src/components/RelatedItems/OutfitItemsList.jsx (hunk 1 above)
Comment (hangyin2020) [reply client/src/components/RelatedItems/OutfitItemsList.jsx:35]: I tried any but it says any is not a function so I tried some instead. but good suggestion!

Comment (slargman) [reply client/src/components/RelatedItems/OutfitItemsList.jsx:35]: ah yeah I'm always forgetting that it's actually `.some`. good call

File: client/src/components/RelatedItems/OutfitItemsList.jsx [hunk 2]
```diff
@@ -1,26 +1,72 @@
...
+  };
+
+  const handleRemove = (itemId) => {
+    const outfitItemsCopy = [...outfitItems];
+    outfitItemsCopy.forEach((item, index) => {
+      if (item.id === itemId) {
+        outfitItemsCopy.splice(index, index + 1);
+      }
+    });
+    setOutfitItems(outfitItemsCopy);
```
[the remark this answers is not in the document]
Comment (slargman) [reply client/src/components/RelatedItems/OutfitItemsList.jsx:45]: you could do something like
```
if (i !== -1) { setOutfitItems([...outfitItems].splice(i, 1)); }
```

**Comment by slargman** [comment]
Feel free to merge whenever you're ready! You're good to go as far as I'm concerned.

**Comment by slargman** [comment]
Ah this branch got out of sync with `main`. I'll pull in the new changes.

**Pull request merged**